backend-go/lib.go

```diff
@@ -0,0 +1,324 @@
+
+	flag.BoolVar(&devFlag, "development mode", false, "If set, program will start in development mode")
+	flag.BoolVar(&natsFlag, "use NATS", true, "If set, service will use NATS as the messaging bus (default)")
+
+	flag.StringVar(&natsUrlFlag, "NATS url", "", "NATS url of the server the service can connect to")
+	flag.StringVar(&natsUserFlag, "NATS user", "", "NATS user name for the authentication with the server")
+	flag.StringVar(&natsPwdFlag, "NATS password", "", "NATS password for the authentication with the server "+
+		warnDescription)
+
+	flag.StringVar(&configFileFlag, "config file path", "", "file path to the config of the service")
+	flag.StringVar(&configKeyFlag, "config file password", "", "password of the config file "+
+		warnDescription)
+
+	flag.StringVar(&serviceNameFlag, "service name", "", "")
+	flag.StringVar(&dataDirFlag, "data directory path", "", "")
+
+	// we don't really like the default message of the flag package
+	flag.Usage = func() {
+		fmt.Printf("Usage: %s [options] [field_0 ... field_n]\n\nParameters:\n", os.Args[0])
+		flag.PrintDefaults()
+	}
+	flag.Parse()
+
+	// additional arguments that can be used by other parts of the service (although they should register flags
+	// themselves before calling the initialize method if possible! -> makes parsing a lot easier + only one behaviour
+	// can be mimicked by multiple non-flag arguments)
+	otherArgs := flag.Args()
+
+	// prepare output map
+	parsedArgs := map[string]any{
+		"NON_FLAG_ARGS": otherArgs,
+	}
+
+	/// populate parsedArgs with entries that were indeed given
+	for k, v := range map[string]any{
+		"DEV":           devFlag,
+		"NATS":          natsFlag,
+		"NATS_URL":      natsUrlFlag,
+		"NATS_USER":     natsUserFlag,
+		"NATS_PASSWORD": natsPwdFlag,
+		"CONFIG_FILE":   configFileFlag,
+		"CONFIG_KEY":    configKeyFlag,
+		"SERVICE_NAME":  serviceNameFlag,
+		"DATA_DIR":      dataDirFlag,
+	} {
+		if isFlagPassed(k) {
+			parsedArgs[k] = v
+		}
+	}
+
+	return parsedArgs
+}
+
+func isFlagPassed(name string) bool {
+	passed := false
+	flag.Visit(func(f *flag.Flag) {
+		if f.Name == name {
+			passed = true
+		}
+	})
+	return passed
+}
+
+func initializeNats() (*nats.Conn, *TelestionError) {
+	var userInfo nats.Option = nil
+	if len(Config.NatsUsr) != 0 && len(Config.NatsPwd) != 0 {
+		userInfo = nats.UserInfo(Config.NatsUsr, Config.NatsPwd)
+	}
+
+	nc, err := nats.Connect(Config.NatsUrl, userInfo)
+
+	if err != nil {
+		return nil, NewTelestionError(err, fmt.Sprintf("Nats client could not be connected for %s!",
+			Config.ServiceName))
+	}
+
+	// Register health check
+	message := fmt.Sprintf(`{"name":"%s"}`, Config.ServiceName) // set here for not needing to reevaluate each time
+	_, err = nc.Subscribe("__telestion__.health", func(m *nats.Msg) {
+		err := m.Respond([]byte(message))
+		if err != nil {
+			// TODO: is Panicln better here?
+			log.Println("Health message message of ", Config.ServiceName, " could not be sent!")
```

Comment:
RE the comment: I don't think we need a panic here. Not being able to send a message one time shouldn't be an issue.